<a href="https://colab.research.google.com/github/yprice93/credit-card-fraud-detection/blob/main/Credit_Card_Fraud_Detection.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Credit Card Fraud Detection

First, we are going to download the dataset from Kaggle.

In [1]:
! pip install -q kaggle

In [2]:
from google.colab import files

In [8]:
files.upload()

Saving kaggle.json to kaggle.json


{'kaggle.json': b'{"username":"yelinprice","key":"0efadd655241991ce66c95ad3da95c68"}'}

In [9]:
! mkdir ~/.kaggle

mkdir: cannot create directory ‘/root/.kaggle’: File exists


In [10]:
! cp kaggle.json ~/.kaggle/

In [11]:
! chmod 600 ~/.kaggle/kaggle.json

In [13]:
! kaggle datasets download mlg-ulb/creditcardfraud

 70% 46.0M/66.0M [00:00<00:00, 254MB/s]
100% 66.0M/66.0M [00:00<00:00, 258MB/s]


In [14]:
!unzip /content/creditcardfraud.zip

Archive:  /content/creditcardfraud.zip
  inflating: creditcard.csv          


# Import Libraries

In [22]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

# EDA

In [16]:
cc = pd.read_csv('creditcard.csv')

In [17]:
cc.head()

Time        V1        V2        V3        V4        V5        V6        V7  \
0   0.0 -1.359807 -0.072781  2.536347  1.378155 -0.338321  0.462388  0.239599   
1   0.0  1.191857  0.266151  0.166480  0.448154  0.060018 -0.082361 -0.078803   
2   1.0 -1.358354 -1.340163  1.773209  0.379780 -0.503198  1.800499  0.791461   
3   1.0 -0.966272 -0.185226  1.792993 -0.863291 -0.010309  1.247203  0.237609   
4   2.0 -1.158233  0.877737  1.548718  0.403034 -0.407193  0.095921  0.592941   

         V8        V9  ...       V21       V22       V23       V24       V25  \
0  0.098698  0.363787  ... -0.018307  0.277838 -0.110474  0.066928  0.128539   
1  0.085102 -0.255425  ... -0.225775 -0.638672  0.101288 -0.339846  0.167170   
2  0.247676 -1.514654  ...  0.247998  0.771679  0.909412 -0.689281 -0.327642   
3  0.377436 -1.387024  ... -0.108300  0.005274 -0.190321 -1.175575  0.647376   
4 -0.270533  0.817739  ... -0.009431  0.798278 -0.137458  0.141267 -0.206010   

        V26       V27       V28  Amount  Class  
0 -0.189115  0.133558 -0.021053  149.62      0  
1  0.125895 -0.008983  0.014724    2.69      0  
2 -0.139097 -0.055353 -0.059752  378.66      0  
3 -0.221929  0.062723  0.061458  123.50      0  
4  0.502292  0.219422  0.215153   69.99      0  

[5 rows x 31 columns]

In [18]:
print(len(cc.columns))

31


In [19]:
print(len(cc))

284807


There are a total of 31 columns and 284,807 rows. We can use class column as the target variable and other columns as features. All columns except class columns are linear variables.

First, we will check NULLs.

In [21]:
# check for nulls in all columns
null_counts = cc.isnull().sum()

print(null_counts)

Time      0
V1        0
V2        0
V3        0
V4        0
V5        0
V6        0
V7        0
V8        0
V9        0
V10       0
V11       0
V12       0
V13       0
V14       0
V15       0
V16       0
V17       0
V18       0
V19       0
V20       0
V21       0
V22       0
V23       0
V24       0
V25       0
V26       0
V27       0
V28       0
Amount    0
Class     0
dtype: int64


Create a matrix of scatter plots. The scatter matrix plots each variable against all the other variables in the DataFrame, resulting in a grid of plots showing the relationships between each pair of variables.

In [ ]:
# create a scatter matrix
pd.plotting.scatter_matrix(cc, figsize=(15,15))
plt.show()